### PR TITLE
PWM refactoring

### LINKED ISSUE #328: `get_polarity` should return Polarity enum
`get_polarity` should return Polarity enum. Currently, it returns the enum value.

### PR DESCRIPTION
**Pull request by jarell-santella**
Refactoring PWM to change frequency and duty cycle set methods to accept float instead of integer (mostly documentation/docstring change). For polarity it now returns the enum.

Closes #328 and #329.

### CODE REVIEW

File: src/edgepi/peripherals/pwm.py [hunk 1]
```diff
@@ -74,33 +75,36 @@ def set_duty_cycle_pwm(self, duty_cycle: float = None):
...
-            polarity (str): "Normal" or "Inversed"
+            polarity (Polarity)
         Return:
             N/A
         """
-        self.pwm.polarity = polarity
+        if polarity == Polarity.NORMAL:
+            self.pwm.polarity = "normal"
+        elif polarity == Polarity.INVERSED:
+            self.pwm.polarity = "inversed"
```
Comment (farzadpanahi) [inline src/edgepi/peripherals/pwm.py:97]: else raise value error

File: src/edgepi/peripherals/pwm.py [hunk 2]
```diff
@@ -74,33 +75,36 @@ def set_duty_cycle_pwm(self, duty_cycle: float = None):
...
-        get pwm polarity
+        Get pwm polarity
         Args:
             N/A
         Return:
-            polarity (str): "Normal" or "Inversed"
+            polarity (Polarity)
         """
-        return self.pwm.polarity
+        return Polarity[self.pwm.polarity.upper()]
```
Comment (farzadpanahi) [inline src/edgepi/peripherals/pwm.py:107]: I think this is fragile. do explicit if elfi else.

**Review by farzadpanahi — APPROVED** [review]

File: src/edgepi/peripherals/pwm.py (hunk 1 above)
Comment (jarell-santella) [reply src/edgepi/peripherals/pwm.py:97]: Done. https://github.com/EdgePi-Cloud/edgepi-python-sdk/pull/330/commits/19b1b9b77905dac49c2241f6f08694f5f38efe71

File: src/edgepi/peripherals/pwm.py (hunk 2 above)
Comment (jarell-santella) [reply src/edgepi/peripherals/pwm.py]: Done. https://github.com/EdgePi-Cloud/edgepi-python-sdk/pull/330/commits/19b1b9b77905dac49c2241f6f08694f5f38efe71

File: src/edgepi/peripherals/pwm.py [hunk 3]
```diff
@@ -74,33 +75,43 @@ def set_duty_cycle_pwm(self, duty_cycle: float = None):
...
             N/A
         Return:
-            polarity (str): "Normal" or "Inversed"
+            polarity (Polarity)
         """
-        return self.pwm.polarity
+        if self.pwm.polarity == Polarity.NORMAL.name.lower():
+            return Polarity.NORMAL
+        if self.pwm.polarity == Polarity.INVERSED.name.lower():
+            return Polarity.INVERSED
```
Comment (farzadpanahi) [inline src/edgepi/peripherals/pwm.py:112]: I meant use literal strings to compare to rather than relying on enum names.

**Review by farzadpanahi — DISMISSED** [review]

File: src/edgepi/peripherals/pwm.py (hunk 3 above)
Comment (jarell-santella) [reply src/edgepi/peripherals/pwm.py]: Done. https://github.com/EdgePi-Cloud/edgepi-python-sdk/pull/330/commits/7c263383dca4c0efccc179ba958ad92ea60d00ea

**Review by farzadpanahi — APPROVED** [review]

**Pull request merged**